sktime/base/_base.py

```diff
@@ -565,46 +565,88 @@ def get_fitted_params(self, deep=True):
+        ]
+
+        for plugin in get_fitted_params_from_plugins:
+            fitted_params.update(plugin())
+
+        return fitted_params
+
+    def _gfp_non_nested_plugin(self):
+        """Plugin to get fitted non-nested parameters."""
+        return self._get_fitted_params()
 
-        def sh(x):
-            """Shorthand to remove all underscores at end of a string."""
-            if x.endswith("_"):
-                return sh(x[:-1])
-            else:
-                return x
+    def _gfp_nested_skbase_plugin(self):
+        """Plugin to get fitted nested skbase parameters."""
+        nested_params = {}
 
-        # add all nested parameters from components that are sktime BaseObject
+        # Add all skbase objects
         c_dict = self._components()
         for c, comp in c_dict.items():
             if isinstance(comp, BaseEstimator) and comp._is_fitted:
                 c_f_params = comp.get_fitted_params()
-                c_f_params = {f"{sh(c)}__{k}": v for k, v in c_f_params.items()}
-                fitted_params.update(c_f_params)
+                c_f_params = {
+                    f"{_rm_underscore(c)}__{k}": v for k, v in c_f_params.items()
+                }
+                nested_params.update(c_f_params)
+
+        return nested_params
+
+    def _gfp_nested_sklearn_plugin(self):
+        """Plugin to get fitted nested sklearn parameters."""
+        fitted_params = self._gfp_non_nested_plugin()
 
-        # add all nested parameters from components that are sklearn estimators
-        # we do this recursively as we have to reach into nested sklearn estimators
+        # Add all nested parameters from components that are sklearn estimators
+        # this is to be done recursively as we have to reach into nested sklearn
+        # estimators
         n_new_params = 42
         old_new_params = fitted_params
         while n_new_params > 0:
             new_params = dict()
             for c, comp in old_new_params.items():
                 if isinstance(comp, _BaseEstimator):
                     c_f_params = self._get_fitted_params_default(comp)
-                    c_f_params = {f"{sh(c)}__{k}": v for k, v in c_f_params.items()}
-                    new_params.update(c_f_params)
+                    c_f_params = {
+                        f"{_rm_underscore(c)}__{k}": v for k, v in c_f_params.items()
+                    }
             fitted_params.update(new_params)
             old_new_params = new_params.copy()
             n_new_params = len(new_params)
 
         return fitted_params
 
+    def _gfp_sklearn_pipeline_plugin(self):
+        """Plugin to get fitted nested sklearn parameters."""
+        from sklearn.pipeline import Pipeline
+
+        _fitted_params = self._gfp_non_nested_plugin()
+        fitted_params = {}
+
+        for _, entity in _fitted_params.items():
+            if isinstance(entity, Pipeline):
+                for name, step in entity.named_steps.items():
```

Comment:
> Can you kindly explain where, and why, any of the names in get_fitted_params are changing? Possibly with an example? I cannot match this statement with the code currently on this PR.

From my first look at the tests, the error specifically mentions `linearregression__n_features_in` as opposed to just `regressor__n_features_in`. It is happening because of calls to `named_steps()`. But I'll look more into it to be sure my suspicion is indeed correct.

> FAILED sktime/base/tests/test_base_sktime.py::test_get_fitted_params_sklearn_nested - AssertionError: assert 'regressor__n_features_in' in dict_keys(['regressor', 'standardscaler__mean', 'standardscaler__n_features_in', 'standardscaler__n_samples_seen', 'st...nearregression__intercept', 'linearregression__n_features_in', 'linearregression__rank', 'linearregression__singular'])

Reply:
let's check carefully whether the issue is with the test, or with the refactor.

This is a very important user facing interface point, and even minor spelling differences can break dependent code.

Let me know if I can help with diagnosing.

Reply:
I was wrong, the error was stemming from a missing dictionary update and has since been fixed.